src/lookup.py

```diff
@@ -855,25 +1068,140 @@ def lookupOne(
+                self.path = path
+                self.filetype = filetype
+                self.ignorewith = ignorewith
+
+        def __init__(self, node, versions):
+            self.invalid = False
+            self.name = node.get("Name")
+            if self.name is None:
+                printf(f"Version {node.attrib} does not have a Name")
+
+            self.toload = []
+
+            for dataNode in node:
+                if dataNode.tag == "version":
+                    name = dataNode.get("Name")
+                    if name is None:
+                        printf(
+                            f"Cannot load version from node {node.attrib} without Name"
+                        )
+                        self.invalid = True
+                    elif name not in versions:
+                        printf(
+                            f"Cannot load version from node {node.attrib} because it is not defined"
+                        )
+                        self.invalid = True
+                    else:
+                        self.toload.append(versions[name])
+                elif dataNode.tag in ("entities", "stages", "roomtypes"):
+                    path = dataNode.get("File")
+                    if path:
+                        self.toload.append(
+                            self.DataFile(
+                                dataNode.tag, path, dataNode.get("IgnoreWith")
+                            )
+                        )
+                    else:
+                        printf(
+                            f"Cannot load file from node {node.attrib} without File path"
+                        )
+                        self.invalid = True
+                else:
+                    printf(
+                        f"Cannot load unknown file type {dataNode.tag} from node {node.attrib}"
+                    )
+                    self.invalid = True
+
+        def allFiles(self, versions=None):
+            files = []
+
+            if versions is None:
+                versions = {}
+
+            versions[self.name] = True
+            for entry in self.toload:
+                if isinstance(entry, MainLookup.Version):
+                    files.extend(entry.allFiles(versions))
+                else:
+                    files.append(entry)
+
+            files = list(
+                filter(
+                    lambda file: file.ignorewith is None
+                    or file.ignorewith not in versions,
+                    files,
+                )
+            )
+
+            return files
+
+    def __init__(self, version):
+        self.stages = StageLookup(version, self)
+        self.roomTypes = RoomTypeLookup(version, self)
+        self.entities = EntityLookup(version, self)
+        self.version = version
+
+        self.loadXML(loadXMLFile("resources/Versions.xml"))
+
+    def loadFromMod(self, modPath, brPath, name, autogenerateContent):
+        versionsPath = os.path.join(brPath, "VersionsMod.xml")
```

Comment:
do we want to commit to supporting this? What does this buy us?

Reply:
VersionsMod.xml allows a mod to use multiple entities / roomtypes / stage files for organization like BR does, as well as to have a different set of entities on different game versions. It's also just nice to have a consistent method of loading all data that both BR and mods use.